File: android/app/src/main/java/com/hungama/starme/ui/theme/Type.kt
@file:OptIn(androidx.compose.ui.text.ExperimentalTextApi::class)

package com.hungama.starme.ui.theme

import androidx.compose.material3.Typography
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.Font
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontVariation
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.em
import androidx.compose.ui.unit.sp
import com.hungama.starme.R

/**
 * Type system.
 *
 * The demo uses two typefaces:
 *   - Display: **Anton** (condensed, heavy, all-caps) — titles, poster billing, numerals.
 *   - Body: **Inter** — everything else.
 *
 * This build ships with *system-font stand-ins* so it compiles and runs with no
 * bundled binaries. To match the demo pixel-for-pixel, the dev team does a
 * one-line swap (see README → "Fonts"):
 *
 *   1. Drop `Anton-Regular.ttf` and `Inter-*.ttf` into `app/src/main/res/font/`.
 *   2. Replace the two families below with `FontFamily(Font(R.font.anton))` etc.
 *
 * Alternatively wire the downloadable Google Fonts provider (also in README).
 */

// Display: Anton (single weight — mapped across the weights the styles request).
val DisplayFontFamily: FontFamily = FontFamily(
    Font(R.font.anton_regular, weight = FontWeight.Normal),
    Font(R.font.anton_regular, weight = FontWeight.Bold),
    Font(R.font.anton_regular, weight = FontWeight.Black),
)

// Body: Inter variable font, weight axis pinned per declared weight.
val BodyFontFamily: FontFamily = FontFamily(
    Font(R.font.inter_variable, weight = FontWeight.Normal, variationSettings = FontVariation.Settings(FontVariation.weight(400))),
    Font(R.font.inter_variable, weight = FontWeight.Medium, variationSettings = FontVariation.Settings(FontVariation.weight(500))),
    Font(R.font.inter_variable, weight = FontWeight.SemiBold, variationSettings = FontVariation.Settings(FontVariation.weight(600))),
    Font(R.font.inter_variable, weight = FontWeight.Bold, variationSettings = FontVariation.Settings(FontVariation.weight(700))),
)

/** Convenience style for the all-caps condensed display treatment (`.disp` in the demo). */
val DisplayCaps = TextStyle(
    fontFamily = DisplayFontFamily,
    fontWeight = FontWeight.Black,
    letterSpacing = 0.02.em,
)

/** Tabular numerals for the countdown / package episode counts. */
val NumeralStyle = TextStyle(
    fontFamily = DisplayFontFamily,
    fontWeight = FontWeight.Black,
)

val StarTypography = Typography(
    // Hero — "Star in your own Micro Drama"
    displayLarge = TextStyle(
        fontFamily = DisplayFontFamily,
        fontWeight = FontWeight.Black,
        fontSize = 44.sp,
        lineHeight = 43.sp,
        letterSpacing = 0.02.em,
    ),
    // Screen h2
    headlineMedium = TextStyle(
        fontFamily = DisplayFontFamily,
        fontWeight = FontWeight.Black,
        fontSize = 26.sp,
        lineHeight = 30.sp,
        letterSpacing = 0.02.em,
    ),
    titleLarge = TextStyle(
        fontFamily = BodyFontFamily,
        fontWeight = FontWeight.SemiBold,
        fontSize = 18.sp,
        lineHeight = 24.sp,
    ),
    titleMedium = TextStyle(
        fontFamily = BodyFontFamily,
        fontWeight = FontWeight.SemiBold,
        fontSize = 15.sp,
        lineHeight = 20.sp,
    ),
    bodyLarge = TextStyle(
        fontFamily = BodyFontFamily,
        fontWeight = FontWeight.Normal,
        fontSize = 15.sp,
        lineHeight = 22.sp,
    ),
    bodyMedium = TextStyle(
        fontFamily = BodyFontFamily,
        fontWeight = FontWeight.Normal,
        fontSize = 14.sp,
        lineHeight = 21.sp,
    ),
    bodySmall = TextStyle(
        fontFamily = BodyFontFamily,
        fontWeight = FontWeight.Normal,
        fontSize = 12.sp,
        lineHeight = 18.sp,
    ),
    labelLarge = TextStyle(
        fontFamily = BodyFontFamily,
        fontWeight = FontWeight.Bold,
        fontSize = 15.sp,
        lineHeight = 20.sp,
    ),
    labelSmall = TextStyle(
        fontFamily = BodyFontFamily,
        fontWeight = FontWeight.Bold,
        fontSize = 11.sp,
        lineHeight = 16.sp,
        letterSpacing = 0.24.em,
    ),
)

File: src/starme/theme/type.ts
// src/starme/theme/type.ts
// From Type.kt. Two families, no others.
// letterSpacing in Compose is em; RN wants points. points = em * fontSize.
import type { TextStyle } from 'react-native';

export const Display = 'Anton-Regular'; // condensed, heavy, used all caps
export const Body = 'Inter-Variable';

type NamedStyle = TextStyle;

export const type = {
  displayLarge: { fontFamily: Display, fontSize: 44, lineHeight: 43, letterSpacing: 0.88 },
  headlineMedium: { fontFamily: Display, fontSize: 26, lineHeight: 30, letterSpacing: 0.52 },
  headlineSmall: { fontFamily: Display, fontSize: 21, lineHeight: 26, letterSpacing: 0.42 },
  titleLarge: { fontFamily: Body, fontWeight: '600', fontSize: 18, lineHeight: 24 },
  titleMedium: { fontFamily: Body, fontWeight: '600', fontSize: 15, lineHeight: 20 },
  titleSmall: { fontFamily: Body, fontWeight: '600', fontSize: 13, lineHeight: 18 },
  bodyLarge: { fontFamily: Body, fontWeight: '400', fontSize: 15, lineHeight: 22 },
  bodyMedium: { fontFamily: Body, fontWeight: '400', fontSize: 14, lineHeight: 21 },
  bodySmall: { fontFamily: Body, fontWeight: '400', fontSize: 12, lineHeight: 18 },
  labelLarge: { fontFamily: Body, fontWeight: '700', fontSize: 15, lineHeight: 20 },
  labelMedium: { fontFamily: Body, fontWeight: '700', fontSize: 12, letterSpacing: 1.4 },
  labelSmall: { fontFamily: Body, fontWeight: '700', fontSize: 11, lineHeight: 16, letterSpacing: 2.64 },
} as const satisfies Record<string, NamedStyle>;

export type TypeKey = keyof typeof type;
